### DISCUSSION TITLE
Meterpreter Staged Payloads

### DISCUSSION THREAD

**Discussion by sanchious** [lem0nSec/ShellGhost#2] [Q&A]
@lem0nSec Very interesting implementation. Thank you for the work you've done. Have you tried testing this with meterpreter staged payloads such as **windows/x64/meterpreter/reverse_tcp** or **windows/x64/meterpreter/reverse_https**?
I am having troubles getting a fully functioning meterpreter session - it dies.
ex:
**_msf6 exploit(multi/handler) > run

[*] Started reverse TCP handler on 192.168.1.235:443 
[*] Sending stage (336 bytes) to 192.168.1.196
[-] Command shell session 3 is not valid and will be closed
[*] 192.168.1.196 - Command shell session 3 closed._**

Thanks in advance!

**Comment by lem0nSec** [comment]
@sanchious Thank you for opening a discussion. 
Staged payloads break the logic of this implementation since they allocate RWX memory.
